pkg/engine/engine.go

```diff
@@ -955,6 +1042,53 @@ func (e *Engine) verificationOverlapWorker(ctx context.Context) {
 						val = res.Raw
 					}
 
+					// Only check the cache for non-targeted scans.
+					// Targeted scans should always be verified.
+					if e.enableDetectionCache && !isTargetedScan {
+						// Check the cache to see if we've already seen this secret.
+						keyBytes := make([]byte, len(res.Raw)+len(res.RawV2))
+						copy(keyBytes, res.Raw)
+						copy(keyBytes[len(res.Raw):], res.RawV2)
```

Comment:
Is it necessary to use both `Raw` and `RawV2`? Isn't the former redundant of V2 is defined?

Reply:
Thanks for pointing this out. The key should actually be `hash(Raw) + hash(RawV2)`. This is due to an unfortunate decision we made with the enterprise solution, where we store both `Hash` and `HashV2` (with `HashV2` being the hash of an empty string if `RawV2` doesn't exist). 😞  For that reason, I believe we need to use both `Raw` and `RawV2` in the key.

I'll have a few others weigh in to double-check if my understanding is correct.

Reply:
I'm reading this on my phone so it's possible I missed it, but shouldn't the hash key contain the detector type ID?

Reply:
The answer is yes: if a secret overlaps between detectors, it can incorrectly report the wrong result.

**Input**
```
snyk = 6b3ae999-3924-486a-9d3b-32315bf0f76b
coda = 6b3ae999-3924-486a-9d3b-32315bf0f76b
```

**Output**
(Note: I have modified the code to always mark the Coda result as verified and delay handling of the Snyk result so that it uses the cached value.) 
```
✅ Found verified result 🐷🔑
Detector Type: Coda
Decoder Type: PLAIN
Raw result: 6b3ae999-3924-486a-9d3b-32315bf0f76b
File: /tmp/overlap

✅ Found verified result 🐷🔑
Detector Type: Coda
Decoder Type: PLAIN
Raw result: 6b3ae999-3924-486a-9d3b-32315bf0f76b
File: /tmp/overlap
```